domain/application/state/application.go

```diff
@@ -274,13 +273,8 @@ WHERE name = $applicationName.name
 // satisfying [applicationerrors.ApplicationNotFoundError] if the application doesn't exist.
 // If the application still has units, as error satisfying [applicationerrors.ApplicationHasUnits]
 // is returned.
-func (st *ApplicationState) DeleteApplication(ctx context.Context, name string) error {
-	db, err := st.DB()
-	if err != nil {
-		return errors.Trace(err)
-	}
-
-	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+func (st *ApplicationState) DeleteApplication(ctx domain.AtomicContext, name string) error {
```

Comment:
This is an example of incorrect use of a method with an `AtomicContext`. 

Just describing it in plain English shows that it has logic that should be in the service layer.
- Get the UUID for the application name.
- Check that no units exist for it.
- Delete all references.
- Delete the record.